rust/nasl-interpreter/src/built_in_functions/misc.rs

```diff
@@ -62,6 +64,66 @@ pub fn unixtime(_: &str, _: &dyn Sink, _: &Register) -> Result<NaslValue, Functi
     }
 }
 
+/// Compress given data with gzip, when headformat is set to 'gzip' it uses gzipheader.
+pub fn gzip(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, FunctionError> {
+    let data = match register.named("data") {
+        Some(ContextType::Value(NaslValue::Null)) => return Ok(NaslValue::Null),
+        Some(ContextType::Value(x)) => Vec::<u8>::from(x),
+        _ => return Err(FunctionError::new("gzip", ("data").into())),
+    };
+    let headformat = match register.named("headformat") {
+        Some(ContextType::Value(NaslValue::String(x))) => x,
+        _ => "noheaderformat",
+    };
+
+    match headformat.to_string().eq_ignore_ascii_case("gzip") {
+        true => {
+            let mut e = GzEncoder::new(Vec::new(), Compression::default());
+            match e.write_all(&data) {
+                Ok(_) => match e.finish() {
+                    Ok(compress) => Ok(NaslValue::Data(compress)),
+                    Err(_) => return Ok(NaslValue::Null),
+                },
+                Err(_) => return Ok(NaslValue::Null),
+            }
+        }
+        false => {
+            let mut e = ZlibEncoder::new(Vec::new(), Compression::default());
+            match e.write_all(&data) {
+                Ok(_) => match e.finish() {
+                    Ok(compress) => Ok(NaslValue::Data(compress)),
+                    Err(_) => return Ok(NaslValue::Null),
+                },
+                Err(_) => return Ok(NaslValue::Null),
+            }
+        }
+    }
+}
+
+/// uncompress given data with gzip, when headformat is set to 'gzip' it uses gzipheader.
+pub fn gunzip(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, FunctionError> {
+    let data = match register.named("data") {
+        Some(ContextType::Value(NaslValue::Null)) => return Ok(NaslValue::Null),
+        Some(ContextType::Value(x)) => Vec::<u8>::from(x),
+        _ => return Err(FunctionError::new("gzip", ("data").into())),
+    };
+
+    let mut uncompress = ZlibDecoder::new(&data[..]);
+    let mut uncompressed = String::new();
+    match uncompress.read_to_string(&mut uncompressed) {
+        Ok(_) => return Ok(NaslValue::String(uncompressed)),
+        Err(_) => (),
```

Comment:
I think I would put the second try in here so that it is clearer that it is a first try and the second is the aftermath.

Otherwise you need to keep track of the return statement to understand it.